File: FrontEnd/src/types/quizAttemptAnswer/createQuizAttemptAnswerDTO.ts
import { CreateQuizAttemptAnswerChoice } from "../quizAttemptAnswerChoice/createQuizAttemptAnswerChoiceDTO";

export interface CreateQuizAttemptAnswerDTO {
    quizAttemptId?: string;
    snapShotQuestionId: string;
    QuizAttemptAnswerChoices: CreateQuizAttemptAnswerChoice[]
}

File: BackEnd/StudyNest.Common/Models/DTOs/EntityDTO/QuizAttemptAnswer/CreateQuizAttemptAnswerDTO.cs
using StudyNest.Common.Attributes;
using StudyNest.Common.Models.DTOs.EntityDTO.QuizAttemptAnswerChoice;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace StudyNest.Common.Models.DTOs.EntityDTO.QuizAttemptAnswer
{
    public class CreateQuizAttemptAnswerDTO
    {
        public string QuizAttemptId { get; set; }
        [TrimmedRequired]
        public string SnapShotQuestionId { get; set; }
        [Required]
        public List<CreateQuizAttemptAnswerChoice> QuizAttemptAnswerChoices { get; set; }
    }
}
